### PR TITLE
Applies a dark theme, based on a global class

### PR DESCRIPTION
**Pull request by jhoff**
~~I need a little help here, but I did most of the the leg work from a styling perspective.~~

~~According to https://bugs.chromium.org/p/chromium/issues/detail?id=608869, there should now be possible to get the value of the chrome devtool theme via `chrome.devtools.panels.themeName`.~~

~~I'm not familiar enough with chrome devtool development to know how to do it correctly, so hopefully this is an easy one for someone that knows what they're doing.~~

~~Implementation of the dark theme is done by adding a `dark` class to the root `#app` element. See change to line 4 at `src/devtools/App.vue`.~~

Edit: When installed in chrome devtools, now detects the theme automatically. Note: the dark theme doesn't work when running `npm run dev`. You will have to add the `dark` class manually to test that I guess.

screenshot from 2017-02-03 00-09-17
screenshot from 2017-02-03 00-09-50
screenshot from 2017-02-03 00-10-13
screenshot from 2017-02-03 00-16-39

Also, any feedback would be appreciated.

### CODE REVIEW

**Comment by jhoff** [comment]
This is an implementation of #79 and #84

**Comment by posva** [comment]
Cool! Just make sure it's synced with the devtools colour

**Comment by jhoff** [comment]
Yup I used all of the same background and border colors that the dark theme already uses.

**Comment by posva** [comment]
I meant that the theme should be dark only when the devtools are in dark mode too. But you're right to do so, thanks for pointing that out 👍

**Comment by jhoff** [comment]
@posva Right, which is why this is marked as a [WIP]. I mentioned in the description that I needed some help finishing that part up. I don't know anything really about extension development. It seems that we can determine which theme devtools is currently using through the `chrome.devtools.panels.themeName` property, but `chrome.devtools` is undefined within the scope of `App.vue` when running `npm run dev` ( which makes sense ).

I'll dig into it a little more, but in the meantime if anyone has a recommendation on the best way to implement the switch from light to dark, let me know. I just don't have a lot of bandwidth at the moment.

**Comment by jhoff** [comment]
Ok It wasn't as complicated as I made it out to be in my head. This should be ready to go.

image
image

**Comment by posva** [comment]
Thanks a lot for looking into this.
I need to test this locally and will come back to you asap

**Comment by crswll** [comment]
Sweet. I was just about to start working on this. Looks good.

**Comment by posva** [comment]
Sorry for the delay, testing now. I'll let you know 🙂

**Comment by posva** [comment]
ping @yyx990803

**Comment by crswll** [comment]
My only objection, not that it matters, is this is very hard to read. I'd personally find it frustrating if I used the dark theme regularly (only reason I don't is because of the vue-devtools).

screen shot 2017-02-11 at 1 09 39 pm

**Comment by yyx990803** [comment]
Nice work - as @crswll mentioned, I think we should give the text in the state inspection panel more contrast (especially the purple and the gray base text).

**Comment by posva** [comment]
Maybe using a cyan instead of the purple and orange instead of red or something similar to
screen shot 2017-02-11 at 19 25 03

**Comment by yyx990803** [comment]
Yeah, emulating the color scheme of the native devtool would be great.

**Comment by jhoff** [comment]
I'll get this updated in the next day or so as soon as I get back from vacation

**Comment by jhoff** [comment]
image

How's this look?

**Comment by posva** [comment]
Looks good!
Where, in the elements tab did you find those colors?

**Comment by jhoff** [comment]
Mostly from the properties viewer for HTML elements:

image

**Comment by andreiglingeanu** [comment]
Very nice looking!

File: src/devtools/App.vue [hunk 1]
```diff
@@ -60,7 +60,8 @@ export default {
   computed: mapState({
     message: state => state.message,
     tab: state => state.tab,
-    newEventCount: state => state.events.newEventCount
+    newEventCount: state => state.events.newEventCount,
+    isDark: () => chrome.devtools && chrome.devtools.panels.themeName === 'dark'
```
Comment (yyx990803) [inline src/devtools/App.vue]: A few small things:

- The devtools code itself doesn't assume it's always running in Chrome. We need to check `typeof chrome !== 'undefined'` first.

- `isDark` doesn't really need to be a computed property. It's initialized only once and shouldn't be in `mapState` either. Let's move it to `data()` instead.

Comment (jhoff) [inline src/devtools/App.vue]: Hows that look?

**Comment by yyx990803** [comment]
Thank you @jhoff , great work! 👍

**Comment by ackzell** [comment]
Hello, 

I just found out a way to integrate vue-devtools into an [NW.js](https://nwjs.io/) dev project. This makes it awesome to have a Vue dev env targeted for nw!

What I found though, is that the dark theme is not being applied to the NW dev tools once I change the theme. I noticed that it has a different menu for changing themes on the settings pane.

NW (0.24.4):
image

Chrome (60.0.3112.113 (Official Build) (64-bit) ): 
image

I think something on this check isn't being matched in the case of the `chrome` object in NW:
https://github.com/vuejs/vue-devtools/blob/master/src/devtools/App.vue#L57

I forced it to be `true` and rebuilt the extension for my use case, that made it appear dark.  

I acknowledge I don't know enough about NW.js yet nor much about how closely it is a "Chrome" window. Much less about creating extensions! Also know that it is a bit of a stretch to ask for a fix on such a thing, I just thought it would be worth sharing my findings.

Cheers!

**Comment by posva** [comment]
@ackzell If anyone finds out how to fix it, we'll welcome a PR 👍

**Pull request merged**